### PR TITLE
Fixing some issues related to GCC compiler

### PR DESCRIPTION
**Pull request by jlconlin**
This should make it possible to compile with both LLVM and GCC.

### CODE REVIEW

File: src/constants.hpp [hunk 1]
```diff
@@ -4,7 +4,7 @@
 #include "boost/hana.hpp"
 #include "dimwits.hpp"
 
-using namespace boost;
+// using namespace boost;
```
Comment (staleyLANL) [inline src/constants.hpp]: Or maybe just remove this entirely?

**Comment by staleyLANL** [comment]
Regarding the removed semicolons after macro invocations: It doesn't really matter, but I prefer to remove the ; in the *macro itself*, then *use* it in all the macro invocations. That means very slightly longer code, because of the extra characters, but editors tend to auto-indent code better if the semicolons appear at the macro invocations themselves, instead of having those just end with a right parenthesis. At least, that's been my experience.

File: src/constants.hpp (hunk 1 above)
Comment (jlconlin) [reply src/constants.hpp]: Yeah, I forgot to delete that.

**Comment by jlconlin** [comment]
I'm going to leave tags.hpp as is (i.e., no semicolons at every macro invocation). Those lines aren't indented anyway so the presence (or lack thereof) doesn't matter in this case. 

I just pushed a slight change to remove the comment. Let me know if you approve.

**Review by staleyLANL — APPROVED** [review]
Yes, that all looks good!

**Pull request merged**